hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/ITestAbfsOutputStream.java

```diff
@@ -84,4 +88,34 @@ public void testMaxRequestsAndQueueCapacity() throws Exception {
     }
   }
 
+  /**
+   * Verify the passing of AzureBlobFileSystem reference to AbfsOutputStream
+   * to make sure that the FS instance is not eligible for GC.
+   *
+   */
+  @Test
+  public void testAzureBlobFileSystemBackReferenceInOutputStream()
+      throws Exception {
```

Comment:
why not force a GC while a write is in progress and verify that the write continues. maybe same for read

Reply:
I don't think we can explicitly force a GC, in the test that I performed, it required the FS instance to be left alone in a separate thread and after sleeping for some time, we can use `System.gc()` which only points/suggests a GC, which did call the `finalize()`, however, I am not sure if this would work in every system per se, since it would depend on the resources available as well at the time.

Reply:
Yes, System.gc() doesn't always trigger gc but you can call it 5 times in a loop which will increase the chance of triggering gc. 
So I guess better to add the test. The test won't fail even if gc is not called. But if gc gets triggered once in a while we will have some coverage and we can always run the test manually.

Reply:
Sure, have added a new test.